lib/pf/Authentication/Source/RADIUSSource.pm

```diff
@@ -167,15 +180,25 @@ sub check_radius_password {
 =cut
 
 sub match_in_subclass {
-    my ($self, $params, $rule, $own_conditions, $matching_conditions) = @_;
+    my ($self, $params, $rule, $own_conditions, $matching_conditions, $extra) = @_;
     $params->{'username'} = $params->{'stripped_user_name'} if (defined($params->{'stripped_user_name'} ) && $params->{'stripped_user_name'} ne '' && isenabled($self->{'stripped_user_name'}));
     my $username =  $params->{'username'};
+
     foreach my $condition (@{ $own_conditions }) {
         if ($condition->{'attribute'} eq "username") {
             if ( $condition->matches("username", $username) ) {
                 push(@{ $matching_conditions }, $condition);
             }
         }
+        if defined($extra->{attributes}) {
```

Comment:
That would not compile you forgot to put () around your condition. 
Yes switching between go and perl is annoying
Just check if $extra is defined
if (defined $extra) {